docs/data-structures/inordertraversal.mdx

```diff
@@ -0,0 +1,206 @@
+---
+title: Inorder traversal of Binary Tree
+slug: /data-structures/inordertraversal
+---
+
+## Introduction
+
+Binary Tree is a type of tree which have atmost two children or we can say atmost degree 2. So to traverse the binary tree according to Depth first search. We have a traversal algorithm called Inorder traversal.
+
+## Explanation 
+In this type of traversal(Inorder traversal). First we have to traverse the left part then its root then right part respectively. In left part we have to traverse whole left subtree after that root of thet subtree and then whole right subtree. Inorder: (left->root->right). 
+### Example 1
+
+<img src="https://i.imgur.com/f0nHPpz.jpg?1" width="600" height="350"/>
+
+```
+
+ First traverse the left part: Node(22).
+ Again it has left child : Node(11).
+ Now it has no left child
+ Inorder traversal: 11
+ Root of this subtree: 22
+ Inorder traversal: 11,22.
+ Right child of this root: 33
+ Inorder traversal: 11,22,33.
+ Root of this subtree: 44
+ Inorder traversal: 11,22,33,44
+ Right child of this root: 66
+ It has left child: 55
+ Now its left child is Null
+ so Inorder traversal: 11,22,33,44,55
+ Root of this subtree: 66
+ Right child of this root: 77
+ Now our Inorder traversal: 11,22,33,44,55,66,77 !!
+
+```
+# Algorithm
+
+For Inorder traversal, we follow the following algorithm:
+
+* First we have to traverse the Binary tree in this order(left->root->right).
+
+* Left means traverse the whole left subtree until we get left child as null. 
+
+* Root means parent if that particular subtree.
+
+* Right means traverse the whole right subtree in this (left->root->right) order until we get null.
```

Comment:
Make it:

* First, we have to traverse the Binary tree in this order(left->root->right).
* Left means traversing the whole left subtree until we get the left child as null. 
* Root means the parent of that particular subtree.
* Right means traverse the whole right subtree in this (left->root->right) order until we get null.